math/mathmore/inc/Math/GSLNLSMinimizer.h

```diff
@@ -104,35 +104,24 @@ class LSResidualFunc : public IMultiGradFunction {
    }
 
    void FdF (const double * x, double & f, double * g) const override {
-      unsigned int n = NDim();
-      std::copy(x,x+n,fX2.begin());
-      const double kEps = 1.0E-4;
-      f = DoEval(x);
-      for (unsigned int i = 0; i < n; ++i) {
-         fX2[i] += kEps;
-         g[i] =  ( DoEval(&fX2.front()) - f )/kEps;
-         fX2[i] = x[i];
-      }
+      f = fChi2->DataElement(x,fIndex,g);
    }
 
 
 private:
 
    double DoEval (const double * x) const override {
-      return fChi2->DataElement(x, fIndex);
+      return fChi2->DataElement(x, fIndex, nullptr);
    }
 
-   double DoDerivative(const double * x, unsigned int icoord) const override {
-      //return  ROOT::Math::Derivator::Eval(*this, x, icoord, 1E-8);
-      std::copy(x,x+NDim(),fX2.begin());
-      const double kEps = 1.0E-4;
-      fX2[icoord] += kEps;
-      return ( DoEval(&fX2.front()) - DoEval(x) )/kEps;
+   double DoDerivative(const double * /* x */, unsigned int /* icoord */) const override {
+      //this function should not be called by GSL
+      assert(false);
+      return 0;
```

Comment:
If this should really not happen, it's better to throw an exception so that the problem is not only caught in debug builds.